test/unit/subcommands/test_stop_subcommand.py

```diff
@@ -1,167 +1,175 @@
-        child_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1
-
-        self.setup_environment_for_os_kill(main_proc_pid)
-        self.setup_main_and_child_proc(main_proc_pid,main_threshold, child_proc_pid, child_threshold)
-
-        stop_subcommand = StopSubcommand()
-        stop_subcommand._kill_pid_in_file_if_exists('/tmp/pid_file_path.pid')
-
-        self.os_kill_patch.assert_called_with(main_proc_pid, signal.SIGKILL)
-        self.assertEquals(3, self.os_kill_patch.call_count)
+            proc.cmdline.return_value = cmdline
+        proc.children.return_value = child_processes if child_processes else []
+        return proc
 
-    def test_stop_subcommand_kills_child_proc_with_sigkill_if_still_running_after_sigterm(self):
-        main_proc_pid = 9999
-        main_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1
-        child_threshold = NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL
-        child_pid = main_proc_pid + 1
+    def _expect_kill_processes(self):
+        self._mock_os_path_exists.return_value = True
+        self._mock_psutil_pid_exists.return_value = True
 
-        self.setup_environment_for_os_kill(main_proc_pid)
-        self.setup_main_and_child_proc(main_proc_pid, main_threshold, child_pid, child_threshold)
-
-        stop_subcommand = StopSubcommand()
-        stop_subcommand._kill_pid_in_file_if_exists('/tmp/pid_file_path.pid')
-
-        self.os_kill_patch.assert_called_with(child_pid, signal.SIGKILL)
-        self.assertEquals(3, self.os_kill_patch.call_count)
-
-    def test_stop_subcommand_kills_proc_with_sigterm(self):
-        main_proc_pid = 9999
-
-        self.setup_environment_for_os_kill(main_proc_pid)
-        self.patch('psutil.Process').return_value = self.create_process_mock(
-            pid=main_proc_pid,
-            is_running_threshold=NUM_OF_IS_RUNNING_CHECKS_TILL_SIGKILL - 1,
-            cmdline=['python', './main.py', 'master'],
-            children=[]
+        self._master_process = self._create_mock_process(
+            self._fake_master_pid,
+            cmdline=['python', 'main.py', 'master'],
         )
 
-        stop_subcommand = StopSubcommand()
-        stop_subcommand._kill_pid_in_file_if_exists('/tmp/pid_file_path.pid')
+        self._slave_child_process = self._create_mock_process(3333)
+        self._slave_process = self._create_mock_process(
+            self._fake_slave_pid,
+            child_processes=[self._slave_child_process],
+            cmdline=['python', 'main.py', 'slave'],
+        )
 
-        self.os_kill_patch.assert_called_with(main_proc_pid, signal.SIGTERM)
-        self.assertEquals(1, self.os_kill_patch.call_count)
+        self._mock_psutil_process.side_effect = [
+            self._slave_process,
+            self._master_process,
+        ]
 
-    def test_stop_subcommand_kills_proc_with_sigkill(self):
-        main_proc_pid = 9999
+        self._stop_subcommand.SIGTERM_SIGKILL_GRACE_PERIOD_SEC = 0
```

Comment:
Is there any reason why you prefer mocking out `time` here? I like the current one as it exercise more code and explicitly says no grace period, which is exactly what the test wants.